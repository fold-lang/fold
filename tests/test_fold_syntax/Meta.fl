// Code quoting
let x = `a;
let x = `1;
let x = `("abc");
let x = `(f x);
let x = `(1 + 2);
let x = `[1, 2, 3];
let x = `((42));
let x = `(let x = 1);
let x = `(let a = 42; a + 1);

//// Code antiquoting
let int_code = `42;
let x = `(1000 + $int_code);
let x = `(1000 + $(`42))

