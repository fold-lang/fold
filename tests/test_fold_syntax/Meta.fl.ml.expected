let x = Ident (Lower "a")
let x = Const (Int 1)
let x = Scope ("(", (Const (String "abc")), ")")
let x =
  Scope ("(", (Seq (None, [Ident (Lower "f"); Ident (Lower "x")])), ")")
let x =
  Scope ("(", (Seq (None, [Sym "+"; Const (Int 1); Const (Int 2)])), ")")
let x =
  Scope
    ("[", (Seq ((Some ","), [Const (Int 1); Const (Int 2); Const (Int 3)])),
      "]")
let x = Scope ("(", (Scope ("(", (Const (Int 42)), ")")), ")")
let x =
  Scope
    ("(", (Form ("let", [Form ("=", [Ident (Lower "x"); Const (Int 1)])])),
      ")")
let x =
  Scope
    ("(",
      (Seq
         ((Some ";"),
           [Form ("let", [Form ("=", [Ident (Lower "a"); Const (Int 42)])]);
           Seq (None, [Sym "+"; Ident (Lower "a"); Const (Int 1)])])), ")")
let int_code = Const (Int 42)
let x = Scope ("(", (Seq (None, [Sym "+"; Const (Int 1000); int_code])), ")")
let x =
  Scope ("(", (Seq (None, [Sym "+"; Const (Int 1000); Const (Int 42)])), ")")
